### ISSUE TITLE
Incorrect TypeCode encode when the message body is nil

### ISSUE THREAD

**Issue by Gsantomaggio** [Azure/go-amqp#332]
When the body message is `nil` the client should encode `TypeCodeNull` value.

Let's go more in-depth :)! 

The behaviour between .NET [amqpnetlite](https://github.com/Azure/amqpnetlite) and this codec differs. 

----

[amqpnetlite](https://github.com/Azure/amqpnetlite) 

Given a message with the `null` body with the amqpnetlite :

```csharp 
Message nullEmptyMessage = new Message(null);
var buff =nullEmptyMessage.Encode();

[0] = {byte} 0
[1] = {byte} 83 // TypeCodeSmallUlong
[2] = {byte} 119 // TypeCodeAMQPValue 
[3] = {byte} 64 // TypeCodeNull
```
✅  This behaviour is correct.

----

[go-amqp](https://github.com/Azure/go-amqp)

```golang
nilGolangMessage := amqp.NewMessage(nil)
binary, _ := nilGolangMessage.MarshalBinary()

[0] = {byte} 0
1 = {uint8} 83  // TypeCodeSmallUlong
2 = {uint8} 117 // TypeCodeApplicationData
3 = {uint8} 160 // TypeCodeVbin8
4 = {uint8} 0
```
❌ That, I think, is a sort of bug when the body is nil; the client should encode `TypeCodeNull` 

----

💡 Possible Fix

It works with this possible fix https://github.com/Gsantomaggio/go-amqp/blob/null_value/message.go#L124-L133

```golang
nilGolangMessage := amqp.NewMessageWithValue(nil)
binary, _ := nilGolangMessage.MarshalBinary()



0 = {uint8} 0
1 = {uint8} 83 // TypeCodeSmallUlong
2 = {uint8} 119 // TypeCodeAMQPValue 
3 = {uint8} 64 // TypeCodeNull
```
-----

oasis documentation

The problem statement is summarised as follows:
"The client should be able to send a body consisting of a single amqp-value section (https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-messaging-v1.0-os.html#type-amqp-value) which in turn consists of the AMQP null value (https://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#type-null)"

------

Thank you

**Comment by Gsantomaggio** [comment]
We use this in our RabbitMQ AMQP 1.0 client, [here](https://github.com/rabbitmq/rabbitmq-amqp-go-client/pull/1/files#diff-9a1220150e5eb419ac43894bc550189b25a70e2cec34cf2fbd66553c8454e64dR137-R139). 

Thank you

**Comment by jhendrixMSFT** [comment]
Thanks for the detailed bug report, it really helps. I haven't had a chance to dig into this too much yet but was wondering if the current behavior is causing issues/incorrect behavior?

**Comment by Gsantomaggio** [comment]
>but was wondering if the current behavior is causing issues/incorrect behavior?

Yes, unfortunately :) 

Let me explain: 
In RabbitMQ, we released [native support for AMQP 1.0](https://www.rabbitmq.com/blog/2024/08/05/native-amqp) 

There is an implementation REST-like for dealing with resource creation, such as queues, exchanges, and bindings. 

To delete a resource, we send a message with a NULL (`TypeCodeNull`) value.  See [here](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/src/rabbit_amqp_management.erl#L235) 

The  other clients  send `TypeCodeSmallUlong, TypeCodeAMQPValue, TypeCodeNull`  in case of `null/nil` message , so it works correctly: 
 - ✅ [Java AMQP 1.0](https://github.com/rabbitmq/rabbitmq-amqp-java-client)  that uses QPID-proton as base client.
 - ✅ [NET AMQP 1.0](https://github.com/rabbitmq/rabbitmq-amqp-dotnet-client) that uses [amqpnetlite](https://github.com/Azure/amqpnetlite) as base client.
 

The [Go AMQP 1.0 Client](https://github.com/rabbitmq/rabbitmq-amqp-go-client) currently uses this client as a base client, and It is not possible to send `TypeCodeAMQPValue` with a `TypeCodeNull` value ❌. 

That's why I temporarily did this fix here: https://github.com/Gsantomaggio/go-amqp/blob/null_value/message.go#L124-L133.  Please let me know if it makes sense to PR it. Any suggestion is welcome :) 

Thank you
